src/screens/NotificationsPromoSheet/index.tsx

```diff
@@ -62,16 +57,15 @@ export function NotificationsPromoSheetInner({
   const primaryButtonOnPress = React.useCallback(async () => {
     if (notificationsDenied) {
       logger.debug(`[NotificationsPromoSheet]: notifications permissions denied (could be default state)`);
-      const result = await requestNotificationPermissions();
-      if (result.status === perms.RESULTS.BLOCKED) {
+      const status = await requestNotificationPermissions();
+      if (status === perms.RESULTS.BLOCKED) {
         analyticsV2.track(analyticsV2.event.notificationsPromoPermissionsBlocked);
         goBack();
-      } else if (result.status === perms.RESULTS.GRANTED) {
-        // only happens on iOS, Android is enabled by default
+      } else if (status === perms.RESULTS.GRANTED || status === perms.RESULTS.LIMITED) {
```

Comment:
This check is being used in multiple places, we should define one function for it.